libraries/ui-library/src/components/six-input/six-input.tsx

```diff
@@ -276,97 +287,118 @@ export class SixInput {
   @Method()
   async reportValidity() {
-    return this.input.reportValidity();
+    return this.nativeInput?.reportValidity();
   }
 
   /** Checks for validity. */
   @Method()
-  async checkValidity() {
-    return this.input.validity.valid;
+  async checkValidity(): Promise<boolean> {
+    if (this.nativeInput === undefined) {
+      return true;
+    }
+    return this.nativeInput.validity.valid;
   }
 
   /** Sets a custom validation message. If `message` is not empty, the field will be considered invalid. */
   @Method()
   async setCustomValidity(message: string) {
     this.customErrorText = '';
     this.customValidation = message !== '';
-    this.input.setCustomValidity(message);
-    this.invalid = !this.input.checkValidity();
+    if (this.nativeInput !== undefined) {
+      this.nativeInput.setCustomValidity(message);
+      this.invalid = !this.nativeInput.checkValidity();
+    }
   }
 
   /** Returns the native input's validity */
   @Method()
-  async getValidity() {
-    return this.input.validity;
+  async getValidity(): Promise<ValidityState | undefined> {
+    return this?.nativeInput?.validity;
   }
 
   /** Returns the native input's validity */
   @Method()
-  async isValid() {
-    return this.input.validity.valid;
+  async isValid(): Promise<boolean> {
+    if (this.nativeInput === undefined) {
+      return true;
+    }
+    return this.nativeInput.validity.valid;
   }
 
   /** Returns the native input's validationMessage */
   @Method()
   async getValidationMessage() {
-    return this.input.validationMessage;
+    if (this.nativeInput === undefined) {
+      return '';
+    }
+    return this.nativeInput.validationMessage;
   }
 
   /** Resets the formcontrol */
   @Method()
   async reset() {
+    if (this.nativeInput === undefined) {
+      return;
+    }
     this.value = this.defaultValue;
     this.customErrorText = '';
     this.customValidation = false;
-    this.input.setCustomValidity('');
+    this.nativeInput.setCustomValidity('');
     this.invalid = false;
   }
 
-  handleChange() {
-    this.value = this.input.value;
-    this.sixChange.emit();
+  private handleChange() {
+    if (this.nativeInput !== undefined) {
+      this.value = this.nativeInput.value;
+      this.sixChange.emit();
+    }
   }
 
-  handleInput() {
-    this.value = this.input.value;
-    this.sixInput.emit();
+  private handleInput() {
+    if (this.nativeInput !== undefined) {
+      this.value = this.nativeInput.value;
+      this.sixInput.emit();
+    }
   }
 
-  handleInvalid() {
+  private handleInvalid() {
     this.invalid = true;
   }
 
-  handleBlur() {
+  private handleBlur() {
     this.hasFocus = false;
     this.sixBlur.emit();
   }
 
-  handleFocus() {
+  private handleFocus() {
     this.hasFocus = true;
     this.sixFocus.emit();
   }
 
-  handleClearClick(event: MouseEvent) {
+  private handleClearClick(event: MouseEvent) {
+    if (this.nativeInput === undefined) {
```

Comment:
Will probably never happen except in testing automation, but the value should probably be set / reset either way (similar to the watch for value).

Reply:
Yes, that should be fine, rewritten to:
```ts
  this.value = '';
  this.sixClear.emit();
  this.sixInput.emit();
  this.sixChange.emit();
  if (this.nativeInput != null) {
    this.nativeInput.focus();
  }
```

Also did the same on the reset() method